bin/tcp_server.ml

```diff
@@ -0,0 +1,65 @@
+open Lwt
+
+(* Shared mutable counter *)
+let counter = ref 0
+
+let listen_address = Unix.inet_addr_loopback
+let port = 9000
+let backlog = 10
+
+let string_to_char s =
+    char_of_int @@ int_of_string s
+
+let handle_message msg =
+    match msg with
+    | "exit" -> "Bye!"
+    | _ ->
+    let param_list = Str.split_delim (Str.regexp " ") msg in
+    match param_list with
+    | [_;_;_] -> let char_param_list = List.map string_to_char param_list in
+    Client.play_note (Array.of_list char_param_list) ();
+    "Values sent to MIDI to play your note"
+    | _      -> "Invalid Input: Provide 3 parameters"
+
+let rec handle_connection ic oc () =
+    Lwt_io.read_line_opt ic >>=
+    (fun msg ->
+        match msg with
+        | Some msg -> 
+            let reply = handle_message msg in
+            Lwt_io.write_line oc reply >>= handle_connection ic oc
+        | None -> Logs_lwt.info (fun m -> m "Connection closed") >>= return)
+
+let accept_connection conn =
+    Printf.printf "Accepting connection from client\n%!";
```

Comment:
As you use the `Logs` library - it makes the most sense to use that all the time. This way you can control where logging gets logged to in a central place. By using printf, you explicitly log to `stdout`, whereas `Logs` can be setup to log to anything. Also, by only using `Logs` you also get consistent output. 

I don't think it's so important to use `Logs` in this program - so this comment is just to give you a hint about how it's useful in practice. Also - if you want to use it - see in the bottom of `logs.mli` for how to use it - there are always good documentation on Daniel Bünzli libraries, even though sometimes complex.